desktop-widgets/btdeviceselectiondialog.cpp

```diff
@@ -156,40 +157,39 @@ void BtDeviceSelectionDialog::on_save_clicked()
 	QString address = remoteDeviceInfo.address().isNull() ? remoteDeviceInfo.deviceUuid().toString() :
 								remoteDeviceInfo.address().toString();
 	saveBtDeviceInfo(address, remoteDeviceInfo);
-	if (remoteDeviceDiscoveryAgent->isActive()) {
-		// Stop the SDP agent if the clear button is pressed and enable the Scan button
-		remoteDeviceDiscoveryAgent->stop();
-#if defined(Q_OS_WIN)
-		remoteDeviceDiscoveryAgent->wait();
-#endif
-		ui->scan->setEnabled(true);
-	}
+	stopScan();
 
 	// Close the device selection dialog and set the result code to Accepted
 	accept();
 }
 
+void BtDeviceSelectionDialog::on_quit_clicked()
+{
+	stopScan();
+
+	// Close the device selection dialog and set the result code to Rejected
+	reject();
+}
+
+void BtDeviceSelectionDialog::showEvent(QShowEvent *event)
+{
+}
+
 void BtDeviceSelectionDialog::on_clear_clicked()
 {
+	if (remoteDeviceDiscoveryAgent->isActive())
+		stopScan();
+
 	ui->dialogStatus->setText(tr("Remote devices list was cleared."));
 	ui->discoveredDevicesList->clear();
-
-	if (remoteDeviceDiscoveryAgent->isActive()) {
-		// Stop the SDP agent if the clear button is pressed and enable the Scan button
-		remoteDeviceDiscoveryAgent->stop();
-#if defined(Q_OS_WIN)
-		remoteDeviceDiscoveryAgent->wait();
-#endif
-		ui->scan->setEnabled(true);
-	}
 }
 
 void BtDeviceSelectionDialog::on_scan_clicked()
 {
-	ui->dialogStatus->setText(tr("Scanning for remote devices..."));
-	ui->discoveredDevicesList->clear();
-	remoteDeviceDiscoveryAgent->start();
-	ui->scan->setEnabled(false);
+	if (remoteDeviceDiscoveryAgent->isActive())
+		stopScan();
```

Comment:
this seems like a user-visible functionality change. the SCAN button now acts like a toggle, turning scanning on and off...

Reply:
Yes, on Linux scanning is asynchronous. Why stop after a given time? On windows scanning takes a fixed time - but there too, why stop scanning after that?

Reply:
That's an interesting question. As I said in my other comment, I always think of scanning as something that I want to do as little as possible. But then I have a hard time articulating exactly WHY. In the mobile case battery life plays a role, but for the desktop? Hmm. Something to think more about.